src/integration/blockchain/arbitrum/arbitrum-client.ts

```diff
@@ -14,6 +36,144 @@ export class ArbitrumClient extends EvmClient {
     swapTokenAddress: string,
   ) {
     super(http, scanApiUrl, scanApiKey, gatewayUrl, privateKey, dfxAddress, swapContractAddress, swapTokenAddress);
+
+    const { ethGatewayUrl, ethApiKey, ethWalletPrivateKey } = GetConfig().blockchain.ethereum;
+    const ethereumGateway = `${ethGatewayUrl}/${ethApiKey ?? ''}`;
+
+    this.#l1Provider = new ethers.providers.JsonRpcProvider(ethereumGateway);
+    this.#l1Wallet = new ethers.Wallet(ethWalletPrivateKey, this.#l1Provider);
+
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    this.initL2Network();
+  }
+
+  async depositCoinOnDex(amount: number): Promise<string> {
+    const ethBridger = new EthBridger(this.#l2Network);
+
+    const depositTx = await ethBridger.deposit({
+      amount: this.convertToWeiLikeDenomination(amount, 'ether'),
+      l1Signer: this.#l1Wallet,
+      l2Provider: this.provider,
+    } as EthDepositParams);
+
+    return depositTx.hash;
+  }
+
+  async withdrawCoinOnDex(amount: number): Promise<string> {
+    const ethBridger = new EthBridger(this.#l2Network);
+
+    const withdrawTx = await ethBridger.withdraw({
+      amount: this.convertToWeiLikeDenomination(amount, 'ether'),
+      l2Signer: this.wallet,
+      from: this.wallet.address,
+      destinationAddress: this.#l1Wallet.address,
+    });
+
+    return withdrawTx.hash;
+  }
+
+  async depositTokenOnDex(l1Token: Asset, _l2Token: Asset, amount: number): Promise<string> {
+    const erc20Bridge = new Erc20Bridger(this.#l2Network);
+    const contract = this.getERC20ContractForDexL1(l1Token.chainId);
+    const decimals = await contract.decimals();
+
+    const approveTx = await erc20Bridge.approveToken({
+      amount: this.convertToWeiLikeDenomination(amount, decimals),
+      l1Signer: this.#l1Wallet,
+      erc20L1Address: l1Token.chainId,
+    });
+
+    await approveTx.wait();
+
+    const depositTx = await erc20Bridge.deposit({
+      amount: this.convertToWeiLikeDenomination(amount, decimals),
+      erc20L1Address: l1Token.chainId,
+      l1Signer: this.#l1Wallet,
+      l2Provider: this.provider,
+    });
```

Comment:
How long does this approval take? What if the service restarts at this point?

Reply:
as discussed, in the first approach lets keep it like this, ideally to be split into separate actions